cobrautil.go

```diff
@@ -173,24 +173,44 @@ func OpenTelemetryRunE(flagPrefix string, prerunLevel zerolog.Level) CobraRunFun
 
 		provider := strings.ToLower(MustGetString(cmd, prefixed("provider")))
 		serviceName := MustGetString(cmd, prefixed("service-name"))
-
+		endpoint, _ := cmd.Flags().GetString(prefixed("endpoint"))
+		
+		var exporter trace.SpanExporter
+		var err error
+
+		// If endpoint is not set, the clients are configured via the OpenTelemetry environment variables or
+		// default values.
+		// See: https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/otlp/otlptrace#environment-variables
+		// or https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/jaeger#environment-variables
 		switch provider {
 		case "none":
 			// Nothing.
 		case "jaeger":
-			exporter, err := jaeger.New(jaeger.WithCollectorEndpoint(jaeger.WithEndpoint(MustGetString(cmd, prefixed("endpoint")))))
+			if endpoint != "" {
+				exporter, err = jaeger.New(jaeger.WithCollectorEndpoint(jaeger.WithEndpoint(endpoint)))
+			} else {
+				exporter, err = jaeger.New(jaeger.WithCollectorEndpoint())
+			}
```

Comment:
This seems fine for now, but in the future when there could be more options, it might make more sense to do something like

```go
var opts []jaeger.CollectorEndpointOption
if endpoint != "" {
    opts = append(opts, jaeger.WithEndpoint(endpoint))
}
exporter, err := jaeger.New(opts...)
```

This strategy can likely be applied to all of the other providers, too.

Reply:
addressed in c4a5fd98d3397e4d3af51c2a674fe5ec7e2468f3